packages/files/README.md

```diff
@@ -0,0 +1,193 @@
+### Files API
+
+This middleware has endpoints for uploading binaries and sharing previously uploaded files between apps/users.
+
+- **Base path**: `/api/files`
+- **Endpoints**:
+  - `POST /api/files/upload` — upload one or multiple files
+  - `POST /api/files/share` — create a new file record that points to an existing binary (re-share)
+
+### Authentication
+
+- Only authenticated, non-deleted users can call the endpoints. Auth is taken from the app session cookie (same as other Keystone-secured endpoints).
+- Unauthorized or deleted users receive `UNAUTHENTICATED/AUTHORIZATION_REQUIRED`.
+
+### Configuration
+
+Set `FILE_UPLOAD_CONFIG` env var to a JSON string:
+
+```json
+{
+  "clients": {
+    "some-app-internal-id": { "name": "Some app", "secret": "<HS256-signing-secret>" }
+  },
+  "quota": {
+    "user": 100,
+    "ip": 100
+  }
+}
+```
+
+### Uploading files
+
+- Method: `POST`
+- Content-Type: `multipart/form-data`
+- Fields:
+  - `file`: one or multiple file binaries
+  - `meta`: JSON string with:
+    - `dv`: `1`
+    - `sender`: `{ dv: 1, fingerprint: string }`
+    - `authedItemId`: UUID of the current user; must equal the authenticated user id
+    - `appId`: one of keys from `clients`
+    - `modelNames`: non-empty array of Models to which this file should be connected
```

Comment:
Is it optional? If so, you need to remove it from the minimal example

Reply:
It looks like extra security checks. May be move it to optional

Reply:
That's important part of a new API. U should strictly define where u want to connect ur file. Without that u just be able to upload file into the storage and don't use it anymore and anywhere